src/coreclr/jit/fgehopt.cpp

```diff
@@ -1101,15 +1097,15 @@ PhaseStatus Compiler::fgCloneFinally()
         {
             BasicBlock* newBlock = blockMap[block];
             // Jump kind/target should not be set yet
-            assert(newBlock->KindIs(BBJ_NONE));
+            assert(!newBlock->HasJump());
```

Comment:
Shouldn't this be:
```suggestion
            assert(newBlock->KindIs(BBJ_ALWAYS));
```
? Or do you also want:
```
            assert(newBlock->KindIs(BBJ_ALWAYS) && !newBlock->HasJump());
```

Reply:
The second is what I was going for; I'll update it.